features/cellular/framework/API/CellularContext.h

```diff
@@ -172,6 +172,11 @@ class CellularContext : public CellularInterface {
      */
     static CellularContext *get_default_nonip_instance();
 
+    /** Get pointer to CellularDevice instance. May be null if not AT-layer.
+     *
+     *  @return pointer to CellularDevice instance
+     */
+    virtual CellularDevice *get_device() const;
```

Comment:
Should be pure virtual?

Reply:
No, we can't add pure virtual methods to API. If someone has inherited this interface then we break the API.

Reply:
I would suggest it doesn't need to be virtual, even. Derived classes can override in a non-virtual way just to get the more-specific type.

Reply:
removed virtual

Reply:
Even without the virtual, you could still also have

    AT_CellularDevice *AT_CellularContext::get_device() const
    {
        return static_cast<AT_CellularDevice *>(CellularContext::get_device());
    }

To give you the covariant return type.